src/modules/Dashboard/features/Applets/Applets.hooks.ts

```diff
@@ -6,56 +6,84 @@ import {
   GetAppletsParams,
   getWorkspaceAppletsApi,
   getWorkspaceFoldersApi,
+  getWorkspaceFolderAppletsApi,
+  getFilteredWorkspaceAppletsApi,
 } from 'api';
 import { DEFAULT_ROWS_PER_PAGE } from 'shared/components';
 import { useAsync } from 'shared/hooks';
 import { workspaces } from 'shared/state';
+import { getObjectFromList } from 'shared/utils';
+import { getAppletsWithLocalFolders } from './Applets.utils';
 
-export const useApplets = (
-  setRows: Dispatch<SetStateAction<(Folder | Applet)[]>>,
-  expandedFolders: string[],
+export const useAppletsWithFolders = (
+  onChangeApplets: Dispatch<SetStateAction<(Folder | Applet)[]>>,
 ) => {
   const { ownerId } = workspaces.useData() || {};
 
-  const { execute: getWorkspaceFolders } = useAsync(getWorkspaceFoldersApi);
-  const { execute: getWorkspaceApplets } = useAsync(getWorkspaceAppletsApi);
+  const [count, setCount] = useState(0);
+  const [isLoading, setLoading] = useState(true);
+  const [expandedFolders, setExpandedFolders] = useState<string[]>([]);
 
-  const [isLoading, setIsLoading] = useState(true);
+  const { execute: getWorkspaceApplets } = useAsync(getWorkspaceAppletsApi);
+  const { execute: getWorkspaceFolders } = useAsync(getWorkspaceFoldersApi);
+  const { execute: getWorkspaceFolderApplets } = useAsync(getWorkspaceFolderAppletsApi);
+  const { execute: getFilteredWorkspaceApplets } = useAsync(getFilteredWorkspaceAppletsApi);
 
   const fetchData = async (getAppletsParams?: GetAppletsParams) => {
+    setLoading(true);
+
+    const { search } = getAppletsParams?.params ?? {};
+    const getAppletsList = search ? getFilteredWorkspaceApplets : getWorkspaceApplets;
+
     const {
       data: { result: folders },
     } = await getWorkspaceFolders({ ownerId: ownerId! });
     const {
-      data: { result: applets },
-    } = await getWorkspaceApplets(
+      data: { result: applets, count },
+    } = await getAppletsList(
       getAppletsParams || {
-        params: {
-          ownerId,
-          limit: DEFAULT_ROWS_PER_PAGE,
-        },
+        params: { ownerId, limit: DEFAULT_ROWS_PER_PAGE },
       },
     );
 
-    let formattedApplets = [
-      ...folders.map((folder) => ({ ...folder, isFolder: true })),
-      ...applets.map((applet) => ({ ...applet, isFolder: false })),
-    ];
+    setCount(count);
 
-    if (!expandedFolders.length) {
-      setRows(formattedApplets);
+    const appletList =
+      applets?.map((applet) => ({
+        ...applet,
+        isFolder: applet.type === 'folder',
```

Comment:
if it is applicable, let's create a enum for applet.type?

Reply:
fixed